sympy/physics/optics/waves.py

```diff
@@ -246,6 +251,12 @@ def __str__(self):
 
     __repr__ = __str__
 
+    def __mul__(self, float):
+        return TWave(self._amplitude*float, self._frequency, self._phase)
```

Comment:
The name of the variable doesn't matter. You should always name the second argument to these functions `other`. Definitely don't name it `float`, which overwrites a builtin.